repairnator/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java

```diff
@@ -2,42 +2,40 @@
 
 import java.util.List;
 
-import fr.inria.spirals.repairnator.process.ProjectState;
-import fr.inria.spirals.repairnator.process.step.PushIncriminatedBuild;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import fr.inria.spirals.jtravis.entities.Build;
 import fr.inria.spirals.repairnator.RepairMode;
+import fr.inria.spirals.repairnator.process.ProjectState;
 import fr.inria.spirals.repairnator.process.step.AbstractStep;
 import fr.inria.spirals.repairnator.process.step.BuildProject;
 import fr.inria.spirals.repairnator.process.step.CloneRepository;
 import fr.inria.spirals.repairnator.process.step.GatherTestInformation;
+import fr.inria.spirals.repairnator.process.step.InspectPreviousBuild;
 import fr.inria.spirals.repairnator.process.step.TestProject;
 import fr.inria.spirals.repairnator.serializer.AbstractDataSerializer;
 
 public class ProjectInspector4Bears extends ProjectInspector {
 	private final Logger logger = LoggerFactory.getLogger(ProjectInspector4Bears.class);
 	
+	private Build previousBuild;
+	private boolean previousBuildFlag;
+	
 	public ProjectInspector4Bears(Build build, String workspace, List<AbstractDataSerializer> serializers, String nopolSolverPath, boolean push, RepairMode mode) {
 		super(build, workspace, serializers, null, push, mode);
+		this.previousBuildFlag = false;
 	}
 	
 	public void run() {
         AbstractStep firstStep = null;
-
-        this.testInformations = new GatherTestInformation(this);
         
         AbstractStep cloneRepo = new CloneRepository(this);
         AbstractStep buildRepo = new BuildProject(this);
         AbstractStep testProject = new TestProject(this);
-        cloneRepo.setNextStep(buildRepo).setNextStep(testProject).setNextStep(this.testInformations);
-
-        if (this.getPushMode()) {
-            pushBuild = new PushIncriminatedBuild(this);
-            pushBuild.setRemoteRepoUrl(PushIncriminatedBuild.REMOTE_REPO_BEAR);
-            this.testInformations.setNextStep(pushBuild);
-        }
+        this.testInformations = new GatherTestInformation(this);
+        AbstractStep inspectPreviousBuild = new InspectPreviousBuild(this);
+        cloneRepo.setNextStep(buildRepo).setNextStep(testProject).setNextStep(this.testInformations).setNextStep(inspectPreviousBuild);
```

Comment:
Yes, but only in the case that the previous build is a failing one. The pipeline for the previous build depends on the status of the previous build. See line 50 in InspectPreviousBuild.java. I will create another pipeline if the previous build is a passing one.